internal/k8s/validator.go

```diff
@@ -379,6 +400,113 @@ func (l *leader) ValidateServiceImportObj(key string, serviceImport *akov1alpha1
 	return nil
 }
 
+// ValidateSSORuleObj would do validation checks
+// update internal CRD caches, and push relevant ingresses to ingestion
+func (l *leader) ValidateSSORuleObj(key string, ssoRule *akov1alpha2.SSORule) error {
+	var err error
+	fqdn := *ssoRule.Spec.Fqdn
+	foundHost, foundSR := objects.SharedCRDLister().GetFQDNToSSORuleMapping(fqdn)
+	if foundHost && foundSR != ssoRule.Namespace+"/"+ssoRule.Name {
+		err = fmt.Errorf("duplicate fqdn %s found in %s", fqdn, foundSR)
+		status.UpdateSSORuleStatus(key, ssoRule, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+		return err
+	}
+
+	refData := make(map[string]string)
+
+	if ssoRule.Spec.SsoPolicyRef == nil {
+		err = fmt.Errorf("SsoPolicyRef is not specified")
+		status.UpdateSSORuleStatus(key, ssoRule, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+		return err
+	}
+	refData[*ssoRule.Spec.SsoPolicyRef] = "SSOPolicy"
+
+	if ssoRule.Spec.OauthVsConfig != nil {
+		oauthConfigObj := ssoRule.Spec.OauthVsConfig
+
+		if len(oauthConfigObj.OauthSettings) != 0 {
+			for _, profile := range oauthConfigObj.OauthSettings {
+				refData[*profile.AuthProfileRef] = "AuthProfile"
+
+				if profile.AppSettings != nil {
+					clientSecret := *profile.AppSettings.ClientSecret
+					clientSecretObj, err := validateSecretReferenceInSSORule(ssoRule.Namespace, clientSecret)
+					if err != nil {
+						err = fmt.Errorf("Got error while fetching %s secret : %s", clientSecret, err.Error())
+						status.UpdateSSORuleStatus(key, ssoRule, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+						return err
+					}
+					if clientSecretObj == nil {
+						err = fmt.Errorf("specified client secret is empty : %s", clientSecret)
+						status.UpdateSSORuleStatus(key, ssoRule, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+						return err
+					}
+					clientSecretString := string(clientSecretObj.Data["clientSecret"])
+					if clientSecretString == "" {
+						err = fmt.Errorf("clientSecret field not found in %s secret", clientSecret)
+						status.UpdateSSORuleStatus(key, ssoRule, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+						return err
+					}
+				}
+
+				if profile.ResourceServer != nil {
+					if *profile.ResourceServer.AccessType == "ACCESS_TOKEN_TYPE_JWT" && profile.ResourceServer.JwtParams == nil {
```

Comment:
better to add `ACCESS_TOKEN_TYPE_JWT` and `ACCESS_TOKEN_TYPE_OPAQUE` as constants